website/client/src/components/payments/sendGiftModal.vue

```diff
@@ -0,0 +1,638 @@
+          <subscription-options
+            v-show="selectedPage === 'subscription'"
+            class="subscribe-option"
+          />
+
+      <!-- gem block -->
+      <div
+        v-show="selectedPage === 'buyGems'"
+      >
+      <div class="gem-group">
+
+
+      <!-- buy gems with money -->
+        <label v-once>
+          {{ $t('howManyGemsPurchase') }}
+        </label>
+
+        <div class="d-flex flex-row align-items-center justify-content-center">
+          <div
+            class="gray-circle"
+            @click="gift.gems.amount <= 0 ? 0 : gift.gems.amount--"
+            >
+            <div
+              class="icon-negative"
+              v-html="icons.negativeIcon"
+              ></div>
+          </div>
+          <div class="input-group">
+            <div class="input-group-prepend input-group-icon align-items-center">
+              <div
+                class="icon-gem"
+                v-html="icons.gemIcon"
+              ></div>
+            </div>
+            <input
+              id="gemsForm"
+              v-model.number="gift.gems.amount"
+              class="form-control"
+              min="0"
+              max="9999"
+            >
+          </div>
+          <div
+            class="gray-circle"
+            @click="gift.gems.amount++"
+            >
+            <div
+              class="icon-positive"
+              v-html="icons.positiveIcon"
+              ></div>
+          </div>
+          </div>
+
+          <!-- the word "total" -->
+          <div class="buy-gem-total">
+            {{ $t('sendGiftTotal') }}
+          </div>
+
+          <!-- the actual dollar amount -->
+          <div class="buy-gem-amount">
+            <span>
+              {{ $t('sendGiftAmount', {cost: formatter.format(totalGems)}) }}
+            </span>
+        </div>
+
+        <!-- change to sending own gems page -->
+        <div
+          :class="{active: selectedPage === 'ownGems'}"
+          @click="selectPage('ownGems')"
+          class="gem-state-change"
+        >
+          {{ $t('wantToSendOwnGems') }}
+        </div>
+      </div>
+
+        <!-- paying for gems -->
+        <payments-buttons
+          class="payment-buttons"
+          :stripe-fn="() => redirectToStripe({gift, uuid: userReceivingGift._id, receiverName})"
+          :paypal-fn="() => openPaypalGift({
+            gift: gift, giftedTo: userReceivingGift._id, receiverName,
+          })"
+          :amazon-data="{type: 'single', gift, giftedTo: userReceivingGift._id, receiverName}"
+          />
+    </div>
+
+        <!-- send gems from balance -->
+      <div
+        v-show="selectedPage === 'ownGems'"
+        >
+        <div class="gem-group">
+        <label v-once>
+          {{ $t('howManyGemsSend') }}
+        </label>
+        <div class="d-flex align-items-center justify-content-center">
+          <div
+            class="gray-circle"
+            @click="gift.gems.amount <= 0 ? 0 : gift.gems.amount--"
```

Comment:
In the case where `gift.gems.amount <= 0` this executes `0` as code, which I think isn't what you want? I'd replace `? 0` with `? gift.gems.amount = 0`